File: server/src/main/java/cn/wildfirechat/app/jpa/Report.java
package cn.wildfirechat.app.jpa;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@IdClass(UserDayId.class)
@Table(name = "t_report")
public class Report implements Serializable {
	@Id
	@Column(length = 64)
	public String userId;
	
	@Id
	@Column
	public long day;

	@Column(length = 64)
	public String title;

	@Column(length = 2048)
	public String content;

	@Column(length = 2048)
	public String tomorrowPlan;

	@Column(length = 2048)
	public String requirement;

	@Column(length = 2048)
	public String reportTo;

	@Column(length = 2048)
	public String ccTo;

	@Column(length = 2048)
	public String reportToGroup;

	@Column(length = 2048)
	public String ccToGroup;
	
}

File: web/src/model/report.js
export default class Report {
    userId;
    title;
    content;
    tomorrowPlan;
    requirement;
    reportTo = [];
    day = 1653314400000;
    ccTo = [];
    reportToGroup = [];
    ccToGroup = [];


    constructor(userId) {
        this.userId = userId;
    }

    valid() {
        if (!this.content || !this.content.trim()) {
            return {
                code: -1,
                msg: '汇报内容不能为空'
            }
        }

        if (!this.reportTo || this.reportTo.length === 0) {
            return {}
        }
    }
}